### PR TITLE
Flag invalid regex patterns in Substitutions

### LINKED ISSUE #391: Normalize Field substitution patterns should be validated
## Observed behavior

Invalid regexes are accepted as search patterns for substitutions in `NormalizeFieldTransformation`
and result in failures loading GTFS feeds with a generic message "Unknown error encountered while transforming zip file", making difficult to troubleshoot.

image

## Expected behavior

A message indicating an invalid search pattern should be shown.

## Steps to reproduce the problem

1. Create a feed source
2. Add a Normalize Field Transformation (under Settings tab, Feed Transformations)
3. Under substitutions, enter an invalid substitution, e.g. `\Cir\b`.
4. Upload a new GTFS zip file and observe the error message.

## Any special notes on configuration used

MTC config

## Version of datatools-server and datatools-ui if applicable (exact commit hash or branch name)

Recent dev branch with normalize fields functionality.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR fixes #391 by changing `NormalizeFieldTranformation` as follows:
- Invalid substitution search patterns are reported in the ZIP transform job status (see screenshot),
- Substitutions with invalid search patterns are rejected on persistence,
- For UI feedback in PR https://github.com/ibi-group/datatools-ui/pull/687, a `valid` field is added to the returned JSON for each substitution of the transformation. 

To test, use a configuration that enables field normalization transformations.

image

### CODE REVIEW

**Review by landonreed — CHANGES_REQUESTED** [review]
See changes requested in #394

File: src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java [hunk 1]
```diff
@@ -18,6 +21,26 @@
...
+            return true;
+        } catch (PatternSyntaxException pse) {
+            return false;
+        }
+    }
+
+    @BsonIgnore
+    public void setValid(boolean value) {
+        // Does nothing (read-only field).
+    }
```
Comment (evansiroky) [inline src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java:42]: Is this still needed?

Comment (binh-dam-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java:42]: It is needed for transmitting valid status to the UI.

Comment (evansiroky) [reply src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java:42]: Can some JSDoc be added about how this prevents an error from occurring upon save?

**Review by evansiroky — APPROVED** [review]
Approved pending resolution of https://github.com/ibi-group/datatools-server/pull/392#discussion_r653943317.

File: src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/transform/Substitution.java]: Added JavaDoc regarding this setter in https://github.com/ibi-group/datatools-server/pull/392/commits/64b206734bf0751839c328ffe52ccec07a5b2d27.

**Review by landonreed — APPROVED** [review]
LGTM

**Pull request merged**